packages/api/src/routes/medical/patient.ts

```diff
@@ -214,6 +214,30 @@ router.get(
   })
 );
 
+/** ---------------------------------------------------------------------------
+ * GET /patient/:id/consolidated/query/:requestId
+ *
+ * Returns the status and information on a specific data query for a given patient.
+ *
+ * @param req.param.id The ID of the patient whose data is to be returned.
+ * @param req.param.requestId The ID of the query to be returned.
+ * @returns the status and information on a specific data query for a given patient.
+ */
+router.get(
+  "/consolidated/query/:requestId",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const { patient } = getPatientInfoOrFail(req);
+    const requestId = getFrom("params").orFail("requestId", req);
+    const query = patient.data.consolidatedQueries?.find((q: ConsolidatedQuery) => q.requestId === requestId);
+    if (!query) throw new NotFoundError("Document query not found");
```

Comment:
Saw that NotFoundError was marked as deprecated - curious if I should be using something else.

Reply:
Oh, if you check the deprecation message it should indicate what do use instead. In this case, it prob says to use `@metriport/core`'s, which is also deprecated and says to use `@metriport/shared`'s 😄

Reply:
BTW, the message should prob say something like "Consolidated query not found", those are different things (document query and consolidated query).